src/Common/src/Interop/Unix/libc/Interop.getcwd.cs

```diff
@@ -10,7 +10,57 @@ internal static partial class Interop
 {
     internal static partial class libc
     {
+        internal static unsafe string getcwd()
+        {
+            const int StackLimit = 256;
+
+            // First try to get the path into a buffer on the stack
+            byte* stackBuf = stackalloc byte[StackLimit];
+            string result = getcwd(stackBuf, StackLimit);
+            if (result != null)
+                return result;
+
+            // If that was too small, try increasing large buffer sizes
+            // until we get one that works or until we hit MaxPath.
+            int maxPath = Interop.libc.MaxPath;
+            int bufferSize = StackLimit;
+            while ((bufferSize = Math.Min(bufferSize * 2, maxPath)) < maxPath)
+            {
+                byte[] buf = new byte[bufferSize];
+                fixed (byte* ptr = buf)
+                {
+                    result = getcwd(ptr, bufferSize);
+                    if (result != null)
+                        return result;
+                }
+            }
+
+            // If we couldn't get the cwd with a MaxPath-sized buffer, something's wrong.
+            throw Interop.GetExceptionForIoErrno(Interop.Errors.ENAMETOOLONG);
+        }
+
+        private static unsafe string getcwd(byte* ptr, int bufferSize)
+        {
+            // Call the real getcwd
+            byte* result = getcwd(ptr, (IntPtr)bufferSize);
```

Comment:
Since the method you're calling wants a size_t (which happens to be defined via a using to IntPtr), I'd cast to `size_t` here in the two places in this method that currently cast to `IntPtr`.